From 8ed26856914fd254ebf8e4b90790e5c97bb9b72c Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Wed, 31 Jan 2024 17:42:55 +0100
Subject: [PATCH 11/20] winegstreamer: Temporarily put pipeline into PLAYING
 state while waiting for the no-more-pads callback.

Some elements (RTSP) won't send no-more-pads in READY/PAUSED state.
---
 dlls/winegstreamer/wg_parser.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/dlls/winegstreamer/wg_parser.c b/dlls/winegstreamer/wg_parser.c
index b76a095ce41..92f4081805e 100644
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -2044,8 +2044,22 @@ static NTSTATUS wg_parser_connect(void *args)
 
     pthread_mutex_lock(&parser->mutex);
 
+    if (parser->uri)
+    {
+        gst_element_set_state(parser->container, GST_STATE_PLAYING);
+        ret = gst_element_get_state(parser->container, NULL, NULL, -1);
+        if (ret == GST_STATE_CHANGE_FAILURE)
+            goto out;
+    }
     while (!parser_no_more_pads(parser) && !parser->error)
         pthread_cond_wait(&parser->init_cond, &parser->mutex);
+    if (parser->uri)
+    {
+        gst_element_set_state(parser->container, GST_STATE_PAUSED);
+        ret = gst_element_get_state(parser->container, NULL, NULL, -1);
+        if (ret == GST_STATE_CHANGE_FAILURE)
+            goto out;
+    }
     if (parser->error)
     {
         if (!parser->use_mediaconv && parser->init_gst == decodebin_parser_init_gst)
-- 
2.44.0

